processing/src/main/java/io/druid/query/filter/SelectorDimFilter.java

```diff
@@ -88,21 +89,50 @@ public Filter toFilter()
       return new SelectorFilter(dimension, value);
     } else {
       final String valueOrNull = Strings.emptyToNull(value);
-      final Predicate<String> predicate = new Predicate<String>()
+      final Long valueAsLong = Longs.tryParse(value);
```

Comment:
Similar comment to the IN filter, it might be worth only doing this if a long predicate is actually requested.